modules/ROOT/pages/clauses/filter.adoc

```diff
@@ -0,0 +1,221 @@
+{
+  "propname": "age"
+}
+----
+
+.Filter on a dynamically computed node property
+[source, cypher]
+----
+MATCH (n:Person)
+FILTER n[$propname] > 40
+RETURN n.name AS name, n.age AS age
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| name | age
+
+| "Lisa" | 48
+
+2+d|Rows: 1
+|===
+
+[[filter-with-where]]
+== `FILTER` as a substitute for `WITH * WHERE`
+
+Unlike `WHERE`, which relies on `MATCH`, `OPTIONAL MATCH`, or `WITH` to define its scope, `FILTER` -- being a clause rather than a subclause -- can filter queries independently of these clauses.
+This can make some queries more concise.
+
+For example, the following two queries are equivalent:
+
+.Filter using `WITH * WHERE`
+[source, cypher]
+----
+UNWIND [1, 2, 3, 4, 5, 6] AS x
+WITH x
+WHERE x > 2
+RETURN x
+----
+
+.Filter using `FILTER`
+[source, cypher]
+----
+UNWIND [1, 2, 3, 4, 5, 6] AS x
+FILTER x > 2
+RETURN x
+----
+
+As such, `FILTER` can be seen as a substitute for the `WITH * WHERE <predicate>` constructs in Cypher.
+
+.Using `FILTER` instead of `WITH * WHERE` in `LOAD CSV`
+=====
+
+The following two xref:clauses/load-csv.adoc[`LOAD CSV`] commands are equivalent:
+
+.companies.csv
+[source, csv, filename="companies.csv"]
+----
+Id,Name,Location,Email,BusinessType
+1,Neo4j,San Mateo,contact@neo4j.com,P
+2,AAA,,info@aaa.com,
+3,BBB,Chicago, info@ ,G
+,CCC,Michigan,info@ccc.com,G
+----
+
+.`LOAD CSV` using `WITH * WHERE`
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///companies.csv' AS row
+WITH row
+WHERE row.Id IS NOT NULL
+MERGE (c:Company {id: row.Id})
+----
+
+.`LOAD CSV` using `FILTER`
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'file:///companies.csv' AS row
+FILTER row.Id IS NOT NULL
+MERGE (c:Company {id: row.Id})
+----
+
+=====
+
+[[limitations]]
+== Limitations
+
+While `FILTER` can act as a substitute for `WITH * WHERE <predicate>` constructs, it does not include the ability of `WITH` to manipulate the variables in scope for subsequent clauses.
+Nor can `FILTER` alias or create new variables.
+In other words, `FILTER` only has the function of `WITH * WHERE <predicate>` and not `WITH <selectedVariable> AS <newVariableName> WHERE <predicate>`.
+
+Additionally, unlike `WHERE`, `FILTER` cannot be used to add filters to patterns.
+
+.`WHERE` inside a node pattern
+[source, cypher]
+----
+WITH 35 AS minAge
+MATCH (a:Person WHERE a.name = 'Andy')-[:KNOWS]->(b:Person WHERE b.age > minAge)
+RETURN b.name AS name`
+----
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| name
+
+| "Timothy"
+
+1+d|Rows: 1
+|===
+
+.Not allowed -- `FILTER` inside a pattern
+[source, cypher, role=test-fail]
+----
+WITH 35 AS minAge
+MATCH (a:Person FILTER a.name = 'Andy')-[:KNOWS]->(b:Person FILTER b.age > minAge)
+RETURN b.name AS name
+----
+
+For more information about how to use `WHERE` in fixed-length and variable-length pattern matching, see xref:clauses/where.adoc#filter-patterns[`WHERE` -> Filter patterns].
```

Comment:
One item you could add is relative to the point the docs makes somewhere that

> When used with MATCH and OPTIONAL MATCH, WHERE adds constraints to the patterns described. It should not be seen as a filter after the matching is finished.

This make a different particularly with `OPTIONAL MATCH`.

```
UNWIND [32,37,40] AS age
OPTIONAL MATCH (p:Person) 
  WHERE p.age = age
RETURN p.name
```
returns
```
╒═══════╕
│p.name │
╞═══════╡
│"Susan"│
├───────┤
│null   │
├───────┤
│"John" │
└───────┘
```
while

```
UNWIND [32,37,40] AS age
OPTIONAL MATCH (p:Person) 
FILTER p.age = age
RETURN p.name
```
returns
```
╒═══════╕
│p.name │
╞═══════╡
│"Susan"│
├───────┤
│"John" │
└───────┘
```